src/ServiceControl.Audit/Infrastructure/DurationRecorder.cs

```diff
@@ -0,0 +1,12 @@
+namespace ServiceControl.Audit;
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+
+record DurationRecorder(Histogram<double> Histogram) : IDisposable
+{
+    readonly Stopwatch sw = Stopwatch.StartNew();
+
+    public void Dispose() => Histogram.Record(sw.ElapsedMilliseconds);
```

Comment:
This is what @danielmarbach means with the ValueStopwatch, the issue isn't using Stopwatch but not use properties that result in too many unneeded conversions and operations. Stopwatch is super slim and was why I mentioned its not needed to use `.Stop`.

Calling ElapsedMilliseconds is enough:

https://github.com/dotnet/runtime/blob/v9.0.1/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Stopwatch.cs#L107-L110

Furhter optimizing really isn't needed for our use case. That would only be needed when measuring microseconds type of durations.